pkgs/development/interpreters/cling/default.nix

```
@@ -155,11 +155,33 @@ let
 
 in
 
-runCommand "cling-${unwrapped.version}" {
+stdenv.mkDerivation {
+  pname = "cling";
+  version = unwrapped.version;
+
   nativeBuildInputs = [ makeWrapper ];
   inherit unwrapped flags;
   inherit (unwrapped) meta;
-} ''
-  makeWrapper $unwrapped/bin/cling $out/bin/cling \
-    --add-flags "$flags"
-''
+
+  dontUnpack = true;
+  dontConfigure = true;
+
+  buildPhase = ''
+    makeWrapper $unwrapped/bin/cling $out/bin/cling \
+      --add-flags "$flags"
+  '';
```

```suggestion
  buildPhase = ''
    runHook preBuild

    makeWrapper $unwrapped/bin/cling $out/bin/cling \
      --add-flags "$flags"

    runHook postBuild
  '';
```

When a phase (like `installPhase`) is overridden, it should start with `runHook preInstall` and end with `runHook postInstall`.

https://github.com/NixOS/nixpkgs/tree/master/pkgs#new-packages